msal/client_application_parameters.go

```diff
@@ -5,26 +5,32 @@ package msal
 
 import "github.com/AzureAD/microsoft-authentication-library-for-go/internal/msalbase"
 
+// TODO(jdoak): determine in the long run if this is even needed. Looks like
+// a hold over from a language conversion.  This doesn't do anything but use
+// applicationCommonParameters, which probably means it has no validity as its
+// own type.
 type clientApplicationParameters struct {
 	commonParameters *applicationCommonParameters
 }
 
-func createClientApplicationParameters(clientID string) *clientApplicationParameters {
-	commonParameters := createApplicationCommonParameters(clientID)
-	p := &clientApplicationParameters{commonParameters}
-	return p
-}
-
-func (p *clientApplicationParameters) setAadAuthority(authorityURI string) {
-	p.commonParameters.setAadAuthority(authorityURI)
+func createClientApplicationParameters(clientID, authorityURI string) (*clientApplicationParameters, error) {
+	// NOTE: I moved setADDAuthority here.  It called a method that was only
+	// used on the output of this function, which is only called here.  It also
+	// ignored the error output. That seems buggy (anytime you ignore an error, must document why).
+	cp, err := createApplicationCommonParameters(clientID, authorityURI)
```

Comment:
>createApplicationCommonParameters [](start = 12, length = 33)

do you mean "removed" and not "moved"? i think it was here because we have different authority types, like AAD, ADFS, or B2C, so eventually, we will need to handle the ADFS and B2C cases (but not in that order). I think what is here is fine, as we'll tackle that issue when we get to it.

Reply:
What I meant (sorry, wasn't clear), was I moved the functionality of setADDAuthority into this func, as it was only used inside this function and was like a single call.

Reply:
I did mean moved, as I moved the setADDAuthority code inside here.  Sorry for the confusion.